File: src/sage-api/yelp/textfiles/test1.js

// there are 1565 items in the categories array
const jsonData= require('./yelp/categories.json'); 
const fsLibrary  = require('fs')

const resto = [];
const food = [];

// this is to get the list of food related category
for( i = 0; i < jsonData.length; i++ ){

    var obj = jsonData[i]

    if (obj.parents == 'restaurants'){
        resto.push(obj.title)
    }

    if (obj.parents == 'food'){
        food.push(obj.title)
        
        fsLibrary.appendFile('foodCat.txt', obj.title, (error) => {
            if (error) throw err;
        })
    }
}


console.log(food)

File: src/sage-api/yelp/textfiles/test1.py
import json
 
# Opening JSON file
f = open('../json_files/categories.json')

data = json.load(f)

# print(data)

# with open('./Restaurants/restoCategory.txt', 'w') as t:
#     for i in data:
#         if 'restaurants' in i['parents']:
#             # print(i['title'])
#             t.write(i['alias'])
#             t.write('\n')


with open('./Restaurants/foodCategory.txt', 'w') as t:
    for i in data:
        if 'food' in i['parents']:
            # print(i['title'])
            t.write(i['alias'])
            t.write('\n')


# with open('allParents.txt', 'w') as t:
#     li = []
#     for i in data:
#         for j in i['parents']:
#             if j not in li:
#                 li.append(j)
    
#     for i in li:
#         t.write(i)
#         t.write('\n')


# THIS WRITES THE ACTIVE CATEGORIES ON ALLACTIVE TXT FILE
# with open('./Activities/allActive.txt', 'w') as t:
#     li = []
#     for i in data:
#         for j in i['parents']:
#             if j not in li and j == 'active':
#                 li.append(i['alias'])


#     for i in li:
#         t.write(i)
#         t.write('\n')
      

# li = []
# for i in data:
#     for j in i['parents']:
#         if j not in li and j == 'active':
#             li.append(i['title'])
            
    # if i['parents'] not in li:
    #     li.append(i['parents'])
    # print(i['parents'])


# Closing file
f.close()

# 